src/azure-cli-testsdk/azure/cli/testsdk/utilities.py

```diff
@@ -45,3 +45,34 @@ def force_progress_logging():
     cmd_logger.removeHandler(test_handler)
     cmd_logger.setLevel(old_cmd_level)
     az_logger.handlers[0].level = old_az_level
+
+
+class StorageAccountKeyReplacer(RecordingProcessor):
+    """Replace the access token for service principal authentication in a response body."""
+
+    def __init__(self, replacement='veryFakedStorageAccountKey'):
+        self._replacement = replacement
+        self._activated = False
+
+    def process_request(self, request):  # pylint: disable=no-self-use
+        import re
+        try:
+            pattern = r"/providers/Microsoft\.Storage/storageAccounts/[^/]+/listKeys$"
+            if re.search(pattern, request.path, re.I):
+                self._activated = True
```

Comment:
Is this used for anything?

Reply:
good catch. this flag is supposed to help skip irrelevant test traffics as i don't like this replacer to sniff everything which would slow down the recording.

Reply:
Cool, this is a pretty good optimization. 

Although do we want to set `self._activated` to false immediately after entering the if statement? Does self._activated work as intended when the except clause is hit?